### PR TITLE
ROE-1995 - TL - Second nationality should be fully optional

### PR DESCRIPTION
**Pull request by mattch1**
### JIRA link
https://companieshouse.atlassian.net/browse/ROE-1995

### Change description

Moved blank checks away from common code and ensured that mandatory and optional fields are treated separately.

### Work checklist

- [x] Tests added where applicable

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/BeneficialOwnerIndividualValidator.java [hunk 1]
```diff
@@ -96,21 +97,26 @@ private boolean validateDateOfBirth(LocalDate dateOfBirth, Errors errors, String
...
         String qualifiedFieldNameSecondNationality = getQualifiedFieldName(OverseasEntitySubmissionDto.BENEFICIAL_OWNERS_INDIVIDUAL_FIELD, BeneficialOwnerIndividualDto.SECOND_NATIONALITY_FIELD);
-        nationalityValidator.validateSecondNationality(
-                qualifiedFieldNameFirstNationality,
-                qualifiedFieldNameSecondNationality,
-                nationality,
-                secondNationality,
-                errors,
-                loggingContext);
-
+        boolean isNotBlank = StringUtils.isNotBlank(secondNationality);
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/BeneficialOwnerIndividualValidator.java:110]: i'd rename the isNotBlank as it doesn't say what is blank, something like isSecondNationalityNotBlank would be better

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/ManagingOfficerIndividualValidator.java [hunk 2]
```diff
@@ -101,17 +102,28 @@ private boolean validateDateOfBirth(LocalDate dateOfBirth, Errors errors, String
...
         String qualifiedFieldNameFirstNationality = getQualifiedFieldName(OverseasEntitySubmissionDto.MANAGING_OFFICERS_INDIVIDUAL_FIELD, ManagingOfficerIndividualDto.NATIONALITY_FIELD);
         String qualifiedFieldNameSecondNationality = getQualifiedFieldName(OverseasEntitySubmissionDto.MANAGING_OFFICERS_INDIVIDUAL_FIELD, ManagingOfficerIndividualDto.SECOND_NATIONALITY_FIELD);
-        var compoundQualifiedFieldName = String.format("%s and %s", qualifiedFieldNameFirstNationality, qualifiedFieldNameSecondNationality);
-        return StringValidators.checkIsNotEqual(nationality, secondNationality, ValidationMessages.SECOND_NATIONALITY_SHOULD_BE_DIFFERENT, qualifiedFieldNameSecondNationality, errors, loggingContext)
-                && StringValidators.isLessThanOrEqualToMaxLength(String.format(CONCATENATED_STRING_FORMAT, nationality, secondNationality), 50, compoundQualifiedFieldName, errors, loggingContext)
-                && StringValidators.isValidCharacters(secondNationality, qualifiedFieldNameSecondNationality, errors, loggingContext);
+        boolean isNotBlank = StringUtils.isNotBlank(secondNationality);
+        if (isNotBlank) {
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/ManagingOfficerIndividualValidator.java:116]: same here for the isNotBlank

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/BeneficialOwnerIndividualValidator.java (hunk 1 above)
Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/BeneficialOwnerIndividualValidator.java]: Not sure we really need to as it is in a method that names the field but have changed it.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/ManagingOfficerIndividualValidator.java (hunk 2 above)
Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/ManagingOfficerIndividualValidator.java]: As above

**Review by markpit — APPROVED** [review]

**Pull request merged**